examples/docker-simple-on-ubuntu-managed-disks/main.tf

```diff
@@ -0,0 +1,196 @@
+    security_group = "${azurerm_network_security_group.network_security_group.id}"
+  }
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_network_interface" "network_interface" {
+  name                = "${var.config["namespace"]}-nic"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  ip_configuration {
+    name                          = "ipconfig1"
+    subnet_id                     = "${azurerm_resource_group.resource_group.id}/providers/Microsoft.Network/virtualNetworks/${var.config["namespace"]}-vnet/subnets/subnet1"
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = "${azurerm_public_ip.public_ip.id}"
+  }
+
+  # Required due to non-Terraform referencing in ip_configuration of subnet from virtual network resource
+  depends_on = ["azurerm_virtual_network.virtual_network"]
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_virtual_machine" "virtual_machine" {
+  name                = "${var.config["namespace"]}-vm"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  network_interface_ids = ["${azurerm_network_interface.network_interface.id}"]
+  vm_size               = "${var.config["vm_size"]}"
+
+  os_profile {
+    computer_name  = "${var.config["namespace"]}-vm"
+    admin_username = "${var.admin_username}"
+    admin_password = ""
+  }
+
+  os_profile_linux_config {
+    disable_password_authentication = true
+
+    ssh_keys {
+      path     = "/home/${var.admin_username}/.ssh/authorized_keys"
+      key_data = "${var.admin_ssh_publickey}"
+    }
+  }
+
+  storage_image_reference {
+    publisher = "${var.config["vm_image_publisher"]}"
+    offer     = "${var.config["vm_image_offer"]}"
+    sku       = "${var.ubuntu_os_version_allowedvalues[var.ubuntu_os_version]}"
+    version   = "latest"
+  }
+
+  storage_os_disk {
+    name              = "osdisk1"
+    managed_disk_type = "${var.config["managed_disk_storage_account_type"]}"
+    create_option     = "FromImage"
+  }
+
+  boot_diagnostics {
+    enabled     = "true"
+    storage_uri = "${azurerm_storage_account.diagnostics_storage_account.primary_blob_endpoint}"
+  }
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_virtual_machine_extension" "docker_extension" {
+  name                = "${var.config["namespace"]}-vmextension"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  virtual_machine_name       = "${azurerm_virtual_machine.virtual_machine.name}"
+  publisher                  = "Microsoft.Azure.Extensions"
+  type                       = "DockerExtension"
+  type_handler_version       = "1.1"
+  auto_upgrade_minor_version = "true"
+  settings                   = "{}"
```

Comment:
(I didn't know it was possible for this field to be optional - as such we should look into making this an optional field so we don't need to specify this - I'll open an issue)